File: InventoryUPDATE001.java
public class Inventory {
    // variables, constructor, and other methods

    public List search(GuitarSpec searchSpec) {
        List matchingGuitars = new LinkedList();
        for (Iterator i = guitars.iterator(); i.hasNext(); ) {
            Guitar guitar = (Guitar) i.next();
            GuitarSpec guitarSpec = guitar.getSpec();

            if (!searchSpec.getBuilder().equals(guitarSpec.getBuilder()))
                continue;

            if ((model != null) && (!model.equals("")) &&
                (!model.equalsIgnoreCase(searchSpec.getModel())))
                continue;

            if (!searchSpec.getType().equals(guitarSpec.getType()))
                continue;

            if (!searchSpec.getBackWood().equals(guitarSpec.getBackWood()))
                continue;

            if (!searchSpec.getTopWood().equals(guitarSpec.getTopWood()))
                continue;
            matchingGuitars.add(guitar);
        }
        return matchingGuitars;
    }
}

File: InventoryUPDATE001.cs
public class Inventory
{
    // variables, constructor, and other methods

    public List<Guitar> Search(GuitarSpec searchSpec)
    {
        List<Guitar> matchingGuitars = new List<Guitar>();
        foreach (Guitar guitar in guitars)
        {
            GuitarSpec guitarSpec = guitar.GetSpec();

            if (!searchSpec.Builder.Equals(guitarSpec.Builder))
                continue;

            if (model != null && !string.IsNullOrEmpty(model) &&
                !model.Equals(searchSpec.Model, StringComparison.OrdinalIgnoreCase))
                continue;

            if (!searchSpec.Type.Equals(guitarSpec.Type))
                continue;

            if (!searchSpec.BackWood.Equals(guitarSpec.BackWood))
                continue;

            if (!searchSpec.TopWood.Equals(guitarSpec.TopWood))
                continue;

            matchingGuitars.Add(guitar);
        }
        return matchingGuitars;
    }
}
